File: AngularLovelyjubbly/ClientApp/src/app/shared/models/turnoverdifferential.ts
import { ITournament } from './tournament';
import { ITeam } from './team';

export interface ITurnoverDifferential {
    turnoverDifferentialId: number;
    tournament: ITournament;
    tournamentShort: number;
    team: ITeam;
    fumbleTakeaways: number;
    interceptionTakeaways: number;
    totalTakeaways: number;
    fumbleGiveaways: number;
    interceptionGiveaways: number;
    totalGiveaways: number;
    turnoverDifferential: number;
}

export class TurnoverDifferential implements ITurnoverDifferential {

    constructor(public turnoverDifferentialId: number = 0,
        public tournament: ITournament = null,
        public tournamentShort: number = null,
        public team: ITeam = null,
        public fumbleTakeaways: number = null,
        public interceptionTakeaways: number = null,
        public totalTakeaways: number = null,
        public fumbleGiveaways: number = null,
        public interceptionGiveaways: number = null,
        public totalGiveaways: number = null,
        public turnoverDifferential: number = null) {
    }
}

File: AngularLovelyjubbly.Data.Sql.Models/TurnoverDifferential.cs
namespace AngularLovelyjubbly.Data.Sql.Models
{
    public partial class TurnoverDifferential
    {
        public TurnoverDifferential()
        {
        }

        public int TurnoverDifferentialId { get; set; }
        //foreign key
        public int TeamId { get; set; }
        //navigation property
        public virtual Team Team { get; set; }
        //foreign key
        public int TournamentId { get; set; }
        //navigation property
        public virtual Tournament Tournament { get; set; }
        public int FumbleTakeaways { get; set; }
        public int InterceptionTakeaways { get; set; }
        public int FumbleGiveaways { get; set; }
        public int InterceptionGiveaways { get; set; }
    }
}
